### PR TITLE
Properly handle unlabeled data

### PR DESCRIPTION
**Pull request by Lguyogiro**
This PR makes Reader/Writers compatible with one another when the input file doesn't contain labels. 

- Updated `featureset.has_labels` to also return `False` when all of the labels are None (this is the case when reading a featureset with `label_col` set to `None`). 
- Added tests to make sure unlabelled data can be read by `CSVReader` and written with `NDJWriter`.

### CODE REVIEW

File: skll/data/featureset.py [hunk 1]
```diff
@@ -414,8 +414,8 @@ def has_labels(self):
             Whether or not this FeatureSet has any finite labels.
         """
         if self.labels is not None:
-            return not (np.issubdtype(self.labels.dtype, float) and
-                        np.isnan(np.min(self.labels)))
+            return not ((np.issubdtype(self.labels.dtype, float) and np.isnan(np.min(self.labels))) or
+                        (all(label is None for label in self.labels)))
```
Comment (desilinguist) [inline skll/data/featureset.py]: Might be better to write this new clause as `np.all(self.labels == True)`.

File: tests/test_featureset.py [hunk 2]
```diff
@@ -972,3 +972,22 @@ def test_featureset_creation_from_dataframe_without_labels_with_vectorizer():
...
+    writer.write()
+    expected_path = os.path.join(_my_dir, "other",
+                                 "featureset_no_labels_expected.jsonlines")
+
+    with open(expected_path) as f_expected:
+        expected_output = f_expected.read()
+    with open(outpath) as f_actual:
+        actual_output = f_actual.read()
+
+    assert_equal(actual_output, expected_output)
```
Comment (desilinguist) [inline tests/test_featureset.py]: Would `eq_` not work here?

Comment (Lguyogiro) [reply tests/test_featureset.py]: It would, I'll update

**Comment by desilinguist** [comment]
Keep it open.

**Comment by aoifecahill** [comment]
keep it open

File: skll/data/featureset.py (hunk 1 above)
Comment (desilinguist) [reply skll/data/featureset.py]: Nope, I think the way you wrote it is actually recommended (and accurate).

**Pull request closed without merging**